src/Game/Game.elm

```diff
@@ -115,7 +115,10 @@ update msg model =
             ( { model | currentScreen = InventoryScreen }, Cmd.none )
 
         InvMsg msg ->
-            ( Inventory.update msg model, Cmd.none )
+            if msg == ExitMsg then
+                ( { model | currentScreen = MapScreen }, Cmd.none )
```

Comment:
Ah, this is a interesting topic and one that I haven't actually broached yet, how to pass msgs from children to parent or whether to do it that way at all! I'm not sure what the answer is, but I thought I'd just point out one thing, I'm trying to remake the existing UI too, see here:
http://www.myabandonware.com/media/screenshots/c/castle-of-the-winds-1ph/castle-of-the-winds_3.gif